archive/type.go

```diff
@@ -6,17 +6,38 @@ import (
 	"github.com/brimsec/zq/zcode"
 	"github.com/brimsec/zq/zdx"
 	"github.com/brimsec/zq/zng"
+	"github.com/brimsec/zq/zng/resolver"
 )
 
 var ErrSyntax = errors.New("syntax/format error encountered parsing zng data")
 
+// TypeIndexer implements the Indexer interface, building an index
+// where each record has one column called "key" whose values
+// are the unique values found in the zbuf.Writer stream
+// from any fields of the incoming records that have the indicated type.
+
 type TypeIndexer struct {
-	Type  zng.Type
-	Table *zdx.MemTable
+	IndexerCommon
+	Type zng.Type
+}
+
+func NewTypeIndexer(path string, refType zng.Type) *TypeIndexer {
+	zctx := resolver.NewContext()
+	return &TypeIndexer{
+		IndexerCommon: IndexerCommon{
+			path:     path,
+			MemTable: zdx.NewMemTable(zctx),
```

Comment:
```suggestion
			MemTable: zdx.NewMemTable(zctx),
			path:     path,
```
To match declaration order.